### ISSUE TITLE
How to do Pilot delivery performance test？

### ISSUE THREAD

**Issue by silencehe09** [istio/istio#10486]
In the Istio 1.0.3 release notes, it shows that "Pilot performance and scalability has been greatly enhanced. Pilot now delivers endpoint updates to 500 sidecars in under 1 second". 

And now we are planning to do some test about Pilot delivery performance , and intend to figure out how the increasement of sidecars number will effect the time Pilot delivers endpoint updates and how many replicas of Pilot pods should we set to gain the best performance of Pilot if there will be thousands of sidecars running in the prod env in the future. The test results are important and critical for our deployment of Istio and capacity planning.

But we run into some difficulties about the test and need some suggetions.
1. How can we evaluate the time between the creation of a traffic rule(VirtualService,DestinationRule,etc) and the finish of Pilot delivers endpoint updates to all sidecars(the last sidecar) in the mesh? Is there any endpoints provided by Pilot or tools or scripts can get the time efficiently? By the way, the command "istioctl proxy-status" can check the syn status, but shows no information about time. 
2. How to set an appropriate replicas of Pilot pods if we have 2000 test sidecars? In the isito deployment yaml file, the hpa of Pilot will autoscale the pods from 1 to 5. Can we increase the replicas(more than 5) to scale the Istio mesh horizonally? 

Look forward to your advice!

@irisdingbj @jnativio

**Comment by zhu0516123** [comment]
@jnativio
Our team are also insterested in  the performance of  information delivery from pilot to envoy. How to test can prove that pilot  delivers endpoint updates to 500 sidecars in under 1 second.

**Comment by suryadu** [comment]
@clyang82 We have recently introduced namespace filtering for Pilot scalability in large topology (https://docs.google.com/document/d/1x8LI3T7SHW-yDrrt3ryEr6iSs9ewMCt4kXtJpIaSSlI/edit#) 
Currently if there are thousands of sidecars in the mesh with the default pilot instances supporting them, we may see performance overhead as each sidecar. Exploiting the namespace filtering will significantly improve scalability here. There are also other config parameters like connectionAge parameter that we recently exposed (https://github.com/istio/istio/issues/7878) that will improve the envoy traffic distribution across Pilot instances which will help the performance. There are few other suggestions as well and we are in the process of updating the documentation with latest information based on our v1.1 performance work. In the meantime, we can share our internal data and guidelines with you directly.
